addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1028,6 +1028,90 @@ registry.HeaderHamburgerFull = publicWidget.Widget.extend({
     },
 });
 
+registry.BottomFixedElement = publicWidget.Widget.extend({
+    selector: '#wrapwrap',
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    start() {
+        this.$target.on('scroll.bottom_fixed_element', _.debounce(this._hideBottomFixedElements.bind(this), 100));
+        $(window).on('resize.bottom_fixed_element', _.debounce(this._hideBottomFixedElements.bind(this), 100));
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this._super(...arguments);
+        this.$target.off('.bottom_fixed_element');
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Hide the element that is fixed at the bottom of the screen if the scroll
+     * reaches the bottom of the page and if this element hides a button.
+     *
+     * @private
+     */
+    _hideBottomFixedElements() {
+        const self = this;
+        const $bottomFixedElements = $('.o_bottom_fixed_element');
+        if ($bottomFixedElements.length) {
+            if ((this.$target[0].offsetHeight + this.$target[0].scrollTop) >= this.$target[0].scrollHeight) {
+                const $buttons = this.$target.find('.btn');
+                _.each($bottomFixedElements, bottomFixedElement => {
+                    $(bottomFixedElement).css('transform', '');
+                    _.each($buttons, button => {
+                        if (self._isColliding($(button), $(bottomFixedElement))) {
+                            $(bottomFixedElement).css('transform', 'translate(0, 100%)');
+                        }
+                    });
```

Comment:
You check the collision with each button for no reason. Don't use each, use a for..of loop (everytime you can anyway)